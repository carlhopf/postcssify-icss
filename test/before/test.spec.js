const browserify = require('browserify');
const rawBody = require('raw-body');
const postcssify2 = require('../../index');
const path = require('path');

describe('before', function () {
	var jsStream;
	var cssStream;
	var onFiles = [];
	
	beforeEach(() => {
		var bundle = browserify();
		
		bundle.plugin(postcssify2, {
			before: [
				require('postcss-nested'),
			],
		});
		
		bundle.add(path.join(__dirname, 'index.js'));
		bundle.on('css stream', (stream) => { cssStream = stream; });
		bundle.on('file', (file) => { onFiles.push(file); });
		
		jsStream = bundle.bundle();
	});
	
	beforeEach((cb) => {
		rawBody(jsStream, (err, buf) => {
			expect(err).toBeFalsy();
			cb(null);
		});
	});
	
	beforeEach((cb) => {
		rawBody(cssStream, (err, buf) => {
			expect(err).toBeFalsy();
			expect(buf.toString('utf8')).toMatchSnapshot();
			cb(null);
		});
	});
	
	test('nested', () => {
		
	});
});

describe('before', function () {
	var jsStream;
	var cssStream;
	var onFiles = [];
	
	beforeEach(() => {
		var bundle = browserify();
		
		bundle.plugin(postcssify2, {
			before: [],
		});
		
		bundle.add(path.join(__dirname, 'index.js'));
		bundle.on('css stream', (stream) => { cssStream = stream; });
		bundle.on('file', (file) => { onFiles.push(file); });
		
		jsStream = bundle.bundle();
	});
	
	beforeEach((cb) => {
		rawBody(jsStream, (err, buf) => {
			expect(err).toBeFalsy();
			cb(null);
		});
	});
	
	beforeEach((cb) => {
		rawBody(cssStream, (err, buf) => {
			expect(err).toBeFalsy();
			expect(buf.toString('utf8')).toMatchSnapshot();
			cb(null);
		});
	});
	
	test('not nested', () => {
		
	});
});
