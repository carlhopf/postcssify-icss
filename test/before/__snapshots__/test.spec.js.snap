// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`before nested 1`] = `
"
  
  .button {
  background-color: red;
}
  
  .button:active {
    background-color: blue;
  }
"
`;

exports[`before not nested 1`] = `
".button {
  background-color: red;
  
  &:active {
    background-color: blue;
  }
}
"
`;
